### PR TITLE
Add Colossalai BertModel and 1d 2d 2.5d 3d tensor parallel configs

### PR DESCRIPTION
**Pull request by Wesley-Jzy**

### CODE REVIEW

File: bert/colossalai_utils/utils.py [hunk 1]
```diff
@@ -22,40 +20,44 @@ def init_w_col(builder):
...
+        cpu_offload = None
 
     rank = get_rank()
     reset_peak_memory_stats(rank)
 
     print_log('Building model')
-    if use_v2:
+    if use_zero:
         shard_strategy = BucketTensorShardStrategy()
         with ZeroInitContext(convert_fp16='fp16' in gpc.config,
```
Comment (feifeibear) [inline bert/colossalai_utils/utils.py]: The latest version colossal has removed the convert_fp16 arg.

File: bert/colossalai_utils/utils.py [hunk 2]
```diff
@@ -67,4 +69,14 @@ def init_w_col(builder):
     lr_scheduler = build_scheduler(len(train_data), optimizer)
     print_log(f'Peak Memory = {max_memory_allocated(rank) / (1024 * 1024)} M')
 
+    if not use_zero:
+        engine, train_data, test_data, _ = colossalai.initialize(model,
```
Comment (feifeibear) [inline bert/colossalai_utils/utils.py]: why don't you use initialize for zero?

File: bert/common/bert.py [hunk 3]
```diff
@@ -78,7 +78,7 @@ def tokenize(examples, mode='concat'):
                                     keep_in_memory=True,
                                     remove_columns='text')
 
-    CONFIG['model']['vocab_size'] = len(tokenizer)
+    CONFIG['model']['vocab_size'] = 50304
```
Comment (feifeibear) [inline bert/common/bert.py]: is this line necessary?

File: bert/common/train.py [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+import time
+
+import torch
+from torch.distributed import all_reduce, get_rank, get_world_size
+from tqdm import tqdm
+
+from zero.common.utils import CONFIG, AsyncMemoryMonitor, print_log, get_tflops
+
+
+def _train(epoch, rank, world_size, train_dataloader, model, criterion, optimizer, lr_scheduler, scaler, mem_monitor):
```
Comment (feifeibear) [inline bert/common/train.py:11]: Can you use typing hit for args?

File: bert/common/bert.py (hunk 3 above)
Comment (Wesley-Jzy) [reply bert/common/bert.py]: It's used for debug and I forget to remove it. Done.

File: bert/common/train.py (hunk 4 above)
Comment (Wesley-Jzy) [reply bert/common/train.py:11]: I will clean all codes from zero benchmark and condense them into example style after verifying the correctness. There are many designs I use now is for benchmark of different sources. They are unnecessary to colossalai-only demo.

File: bert/colossalai_utils/utils.py (hunk 1 above)
Comment (Wesley-Jzy) [reply bert/colossalai_utils/utils.py]: Done.

**Review by feifeibear — APPROVED** [review]

**Pull request merged**